src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -212,7 +232,7 @@ protected override void StartPipelineExecution()
                         {
                             // prepare invoke thread
                             SetupInvokeThread(Thread.CurrentThread, false);
-                            this.InvokeThreadProc();
+                            InvokeThreadProc();
```

Comment:
Same here. Why not use `invokeThreadProcDelegate` here?

Reply:
Actually, I changed my mind about this.  Impersonation flow only needs to be performed when running on a *new* thread.  If the same thread is used, there is no need to run as impersonated regardless of whether the thread contains impersonation.  I feel the code expresses this by always using ThreadProc() when not started on a new thread.

Reply:
Good to know.